### PR TITLE
Improve conda list error when egg-link finds too many egg-infos, fix …

### PR DESCRIPTION
**Pull request by ihnorton**
…#8948

### CODE REVIEW

**Comment by ihnorton** [comment]
Before:
```
[long error message suggesting to submit a bug report to conda]
```
(https://github.com/conda/conda/issues/8948#issuecomment-512743235)


---

After:

setup: manually copied `eql.egg-info` to `eql_qq.egg-info` to recreate the issue

```
conda list                                                                                                                                    ✘ 1

CondaError: Expected exactly one `egg-info` directory in '/Users/inorton/git/conda/src/eql', via egg-link 'lib/python3.7/site-packages/eql.egg-link'. Instead found: ['eql_qq.egg-info', 'eql.egg-info']
```

File: conda/common/pkg_formats/python.py [hunk 1]
```diff
@@ -913,7 +914,10 @@ def get_dist_file_from_egg_link(egg_link_file, prefix_path):
         egg_info_fnames = ()
 
     if egg_info_fnames:
-        assert len(egg_info_fnames) == 1, (egg_link_file, egg_info_fnames)
+        if len(egg_info_fnames) != 1:
+            raise CondaError(
+                    "Expected exactly one `egg-info` directory in '{}', via egg-link '{}'."
+                    " Instead found: {}".format(egg_link_contents, egg_link_file, egg_info_fnames))
```
Comment (msarahan) [inline conda/common/pkg_formats/python.py]:
```suggestion
                    " Instead found: {}.  These are often left over from "
                    "legacy operations that did not clean up correctly.  Please "
                    "remove all but one of these.".format(egg_link_contents, 
                    egg_link_file, egg_info_fnames))
```

**Comment by msarahan** [comment]
Great improvement!  Thank you!  I added more language to hopefully help guide people towards what they should do to fix the problem.

**Comment by ihnorton** [comment]
Updated, thanks.

**Pull request merged**